DataStructuresMemories/Controllers/ControllManager.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  ControllManager.swift
+//  DataStructuresMemories
+//
+//  Created by Oleksii Kolomiiets on 4/26/18.
+//  Copyright © 2018 Oleksii Kolomiiets. All rights reserved.
+//
+
+import Foundation
+
+
+class ATDControlManager: ATDControlProtocol {
+    
+    init(for type: DataType) {
+        typeOfDataStruct = type
+    }
+    
+    var typeOfDataStruct: DataType
+    
+    func createMenu() -> [TypeButton] {
+        var collectionButtons = [TypeButton]()
```

Comment:
return typeOfDataStruct.buttons.map { <your implementation goes here> } is preferrable

Reply:
Got it.